src/tools/tests/manifest2proto-test.ts

```diff
@@ -9,13 +9,259 @@
  */
 import {assert} from '../../platform/chai-web.js';
 import {fs} from '../../platform/fs-web.js';
-import {serialize2proto} from '../manifest2proto.js';
+import {serialize2proto, typeToJSON, manifestToJSON, capabilitiesToProtoOrdinals, schemaToJSON} from '../manifest2proto.js';
+import {EntityType} from '../../runtime/type.js';
+
+import protobuf from 'protobufjs';
+import {Manifest} from '../../runtime/manifest.js';
+import {Capabilities} from '../../runtime/capabilities.js';
+
+const rootNamespace = protobuf.loadSync('./java/arcs/core/data/proto/recipe.proto');
+const manifestProto = rootNamespace.lookupType('arcs.ManifestProto');
+const typeProto = rootNamespace.lookupType('arcs.TypeProto');
+const CAPABILITY_ENUM = rootNamespace.lookupEnum('arcs.Capability');
 
 describe('manifest2proto', () => {
-  it('encodes the example manifest', async () => {
+  // The tests below construct the JSON representation equivalent to the proto,
+  // construct the proto object from the constructed JSON and produce JSON back
+  // from the proto object. This ensures that all JSON produced fits the
+  // expectations of the protobufjs library and the shape of the proto declaration.
+
+  it('encodes a recipe with use, map, create handles, ids and tags', async () => {
+    const manifest = await Manifest.parse(`
+      particle Abc in 'a/b/c.js'
+        a: reads X {a: Text}
+        b: reads Y {b: Number}
+        c: writes Z {c: Boolean}
+
+      recipe
+        a: use #tag1 #tag2
+        b: map 'by-id'
+        c: create persistent
+        Abc
+          a: a
+          b: b
+          c: c
+    `);
+    const fromProto = manifestProto.fromObject(await manifestToJSON(manifest)).toJSON();
```

Comment:
Might be worth defining a helper function for this, since you use it in a few places.

Reply:
Done!